server/py/services/api/crud/model_monitoring/deployment.py

```diff
@@ -2465,6 +2468,60 @@ def _extract_meps_from_model_runner_step(
                 )
         return model_endpoints_instructions
 
+    async def delete_application_records(
+        self, application_name: str, endpoint_ids: typing.Optional[list[str]] = None
+    ) -> None:
+        """
+        Deletes the application records from the model monitoring database.
+        This method is used to delete the records of a specific application.
+
+        :param application_name: The name of the application to delete records for.
+        :param endpoint_ids:     List of endpoint IDs to delete records for.
+        """
+        if endpoint_ids:
+            endpoint_id_list = endpoint_ids
+        else:
+            endpoints_data = (
+                await services.api.crud.ModelEndpoints().list_model_endpoints(
+                    project=self.project,
+                    uids=endpoint_ids,
```

Comment:
`endpoint_ids` is none in that case
```suggestion

```

Reply:
Also, since `endpoint_id_list` is only relevant for non-batch apps, you can move its construction into the if clause starting at L2504. Otherwise, we might call `list_model_endpoints` (if `endpoint_ids=None`) when no need for that.